src/main/scala/beam/utils/map/R5MapStatsCalculator.scala

```diff
@@ -1,24 +1,170 @@
 package beam.utils.map
 
-import com.conveyal.osmlib.OSM
+import com.conveyal.osmlib.{OSM, Way}
 import com.conveyal.r5.point_to_point.builder.TNBuilderConfig
 import com.conveyal.r5.streets.EdgeStore
 import com.conveyal.r5.streets.EdgeStore.EdgeFlag
 import com.conveyal.r5.transit.TransportNetwork
 
+import java.{lang, util}
 import scala.collection.JavaConverters._
+import scala.collection.mutable
 import scala.util.Try
 
 object R5MapStatsCalculator {
 
   def main(args: Array[String]): Unit = {
-//    LoggerFactory.getLogger("com.conveyal").asInstanceOf[Logger].setLevel(Level.INFO)
-    val pathToOsm = args(0)
+    // LoggerFactory.getLogger("com.conveyal").asInstanceOf[Logger].setLevel(Level.INFO)
+    val pathToOsm = "/mnt/data/work/beam/beam-production/production/sfbay/r5-updated/sf-bay.osm.pbf" // args(0)
 
+    analyzeR5Map(pathToOsm)
+    analyzeOSMMap(pathToOsm)
+  }
+
+  private def analyzeOSMMap(pathToOsm: String): Unit = {
+    val osm = new OSM(null)
+    try {
+      osm.readFromFile(pathToOsm)
+
+      println(s"Number of OSM nodes: ${osm.nodes.size()}")
+      println(s"Number of OSM ways: ${osm.ways.size()}")
+
+      printAllTags(osm)
+      printOSMTagInfo(osm)
+    } finally {
+      Try(osm.close())
+    }
+  }
+
+  private def printAllTags(osm: OSM): Unit = {
+    val minFrequency = 1000
+    println(s"All existing tags with the minimum frequency of $minFrequency (the format is <tag>[<frequency>]):")
+    val tagsToCnt = osm.ways.asScala
+      .flatMap {
+        case (_, way) =>
+          way.tags.asScala.map { tag =>
+            tag.key
+          }
+      }
+      .foldLeft(scala.collection.mutable.HashMap.empty[String, Int]) {
```

Comment:
This could be a `foreach` since it is using a mutable collection anyway. In fact I think from a memory perspective the `fold` might end up creating more duplicative objects